tests/test_metric_weighting.py

```diff
@@ -0,0 +1,235 @@
+test_ids = np.hstack(zip(unweight_ids, weight_ids))
+
+class_weight = dict([(i, standard_weight) for i in range(nb_classes)])
+class_weight[weighted_class] = high_weight
+
+sample_weight = np.ones((y_train.shape[0])) * standard_weight
+sample_weight[y_train == weighted_class] = high_weight
+
+temporal_X_train = np.reshape(X_train, (len(X_train), 1, X_train.shape[1]))
+temporal_X_train = np.repeat(temporal_X_train, timesteps, axis=1)
+temporal_X_test = np.reshape(X_test, (len(X_test), 1, X_test.shape[1]))
+temporal_X_test = np.repeat(temporal_X_test, timesteps, axis=1)
+
+temporal_Y_train = np.reshape(Y_train, (len(Y_train), 1, Y_train.shape[1]))
+temporal_Y_train = np.repeat(temporal_Y_train, timesteps, axis=1)
+temporal_Y_test = np.reshape(Y_test, (len(Y_test), 1, Y_test.shape[1]))
+temporal_Y_test = np.repeat(temporal_Y_test, timesteps, axis=1)
+
+temporal_sample_weight = np.reshape(sample_weight, (len(sample_weight), 1))
+temporal_sample_weight = np.repeat(temporal_sample_weight, timesteps, axis=1)
+
+
+def create_sequential_model():
+    model = Sequential()
+    model.add(Dense(32, input_shape=(input_dim,)))
+    model.add(Activation('relu'))
+    model.add(Dense(nb_classes))
+    model.add(Activation('softmax'))
+    return model
+
+
+def create_temporal_sequential_model():
+    model = Sequential()
+    model.add(GRU(32, input_shape=(timesteps, input_dim), return_sequences=True))
+    model.add(TimeDistributedDense(nb_classes))
+    model.add(Activation('softmax'))
+    return model
+
+
+# @keras_test
+def _fit_weights_sequential(model, class_weight=None, sample_weight=None,
+                             X_train=X_train, Y_train=Y_train,
+                             X_test=X_test, Y_test=Y_test):
+    if sample_weight is not None:
+        model.fit(X_train, Y_train, batch_size=batch_size,
+                  nb_epoch=nb_epoch // 3, verbose=0,
+                  class_weight=class_weight, sample_weight=sample_weight)
+        model.fit(X_train, Y_train, batch_size=batch_size,
+                  nb_epoch=nb_epoch // 3, verbose=0,
+                  class_weight=class_weight, sample_weight=sample_weight,
+                  validation_split=0.1)
+        model.fit(X_train, Y_train, batch_size=batch_size,
+                  nb_epoch=nb_epoch // 3, verbose=0,
+                  class_weight=class_weight, sample_weight=sample_weight,
+                  validation_data=(X_train, Y_train, sample_weight))
+    else:
+        model.fit(X_train, Y_train, batch_size=batch_size,
+                  nb_epoch=nb_epoch // 2, verbose=0,
+                  class_weight=class_weight, sample_weight=sample_weight)
+        model.fit(X_train, Y_train, batch_size=batch_size,
+                  nb_epoch=nb_epoch // 2, verbose=0,
+                  class_weight=class_weight, sample_weight=sample_weight,
+                  validation_split=0.1)
+
+
+# no evaluation weights: reference point
+model = create_sequential_model()
```

Comment:
I think you want to all this in `setup_module`

http://doc.pytest.org/en/latest/xunit_setup.html